examples/weiroll.ts

```diff
@@ -0,0 +1,87 @@
+import { solidityPacked } from 'ethers_v6';
+import { ABI_CODER, fnCalldata } from './common';
+
+// see weiroll: https://github.com/weiroll/weiroll/blob/main/README.md
+
+export enum CallType {
+  DelegateCall,
+  Call,
+  StaticCall,
+  CallWithValue,
+}
+
+export const END_OF_ARGS = 0xff;
+
+// encode command args for weiroll
+export const encodeCommand = (
+  selector: string,
+  flags: number,
+  input: bigint,
+  output: number,
+  target: string
+) => {
+  return solidityPacked(
+    ['bytes4', 'uint8', 'uint48', 'uint8', 'address'],
+    [selector, flags, input, output, target]
+  );
+};
+
+// calltype to number encoding
+export const callTypeToNumber = (callType: CallType) => {
+  switch (callType) {
+    case CallType.DelegateCall:
+      return 0;
+    case CallType.Call:
+      return 1;
+    case CallType.StaticCall:
+      return 2;
+    case CallType.CallWithValue:
+      return 3;
+  }
+};
+
+// encode flag part of the weiroll command
+export const encodeFlag = (
+  isTuple: boolean,
+  isExtendedCommand: boolean,
+  callType: CallType
+) => {
+  return (
+    (isTuple ? 0x80 : 0x00) |
+    (isExtendedCommand ? 0x40 : 0x00) |
+    callTypeToNumber(callType)
+  );
+};
+
+// encode the input part of the weiroll command
+export const encodeInput = (
```

Comment:
It would be great to understand what all these are (without specific reference to weiroll docs)